src/proto/h1/encode.rs

```diff
@@ -181,6 +197,22 @@ impl Encoder {
     }
 }
 
+// FIXME: dry up
+fn write_headers(headers: &HeaderMap, dst: &mut Vec<u8>) {
+    for (name, value) in headers {
+        extend(dst, name.as_str().as_bytes());
+        extend(dst, b": ");
+        extend(dst, value.as_bytes());
+        extend(dst, b"\r\n");
+    }
+}
+
+#[inline]
+fn extend(dst: &mut Vec<u8>, data: &[u8]) {
+    dst.extend_from_slice(data);
+}
+// end FIXME: dry up
+
```

Comment:
There is a lot of code in `src/proto/h1/role.rs` that is concerned with the proper casing of headers. I think we should use that same code. I want to pull that code into a new module, something like `src/proto/h1/headers.rs`. Alternative, we should make those functions in `roles.rs` public.

Reply:
i made those functions public